packages/react-components/src/components/SendBox.tsx

```diff
@@ -288,15 +288,21 @@ export const SendBox = (props: SendBoxProps): JSX.Element => {
     [mergedSendIconStyle, onRenderIcon]
   );
 
+  /* @conditional-compile-remove(file-sharing) */
+  // Ensure that errors are cleared when there are no files in sendbox
+  React.useEffect(() => {
+    if (!props.activeFileUploads?.filter((upload) => !upload.error).length) {
```

Comment:
1. Why do you need to filter out files that have upload errors? Is the user allowed to send the message if some of the files failed to upload?

2. You shouldn't need this `useEffect`. A change in `props.activeFileUploads` always causes a rerender, so instead you can change line 303 to be `containsPendingUpload(props.activeFileUploads) && fileUploadsPendingError`.

Reply:
1. Yes. Users are allowed to send message even if there are failed uploads. (Those uploads don't show in sendbox)
2. How do I set `setFileUploadsPendingError(undefined);` using your approach? `fileUploadsPendingError` will always be true

Reply:
Ah right, you actually need to clear internal state whenever `activeFileUploads` becomes empty.